addons/sale_loyalty_delivery/models/sale_order.py

```diff
@@ -4,15 +4,26 @@
 from odoo import _, models
 from odoo.fields import Command
 
+
 class SaleOrder(models.Model):
     _inherit = 'sale.order'
 
+    # delivery overrides
+
+    def _compute_amount_total_without_delivery(self):
+        res = super()._compute_amount_total_without_delivery()
+        return res - sum(
+            self.order_line.filtered(
+                lambda l: l.coupon_id and l.coupon_id.program_type in ['ewallet', 'gift_card']
+            ).mapped('price_unit')
+        )
```

Comment:
Nitpick, but one line less
```suggestion
        return res - sum(self.order_line.filtered(
            lambda l: l.coupon_id and l.coupon_id.program_type in ['ewallet', 'gift_card']
        ).mapped('price_unit')
```

Reply:
Not convinced, it's less clear imho.